File: Fibonacci.py
import numpy as np
def f(x):
    fx = x**2 + (54/x) #Enter your function here
    return fx

def fib(n):

    a=0
    b=1
    if n == 0:
        return a
    for i in range(2,n+1):
        c=a+b
        a=b
        b=c
    return b

def main():
    a = float(input('Enter a: '))
    b = float(input('Enter b: '))
    n = int(input('Enter n: '))
    L = b-a
    print('L = ',L)
    k=2
    while k<=n:
        print(f'------------Iteration - {k+1}--------------')
        I_0 = n-k+1
        I_1 = n+1
        F_0 = fib(I_0+1)
        F_1 = fib(I_1+1)
        Lk = (float(F_0)/float(F_1))*L
        print('I0 = ',I_0)
        print('I1 = ',I_1)
        print('F0 = ',F_0)
        print('F1 = ',F_1)
        print(f'L_{k} = ',Lk)
        x1 = a + Lk
        x2 = b - Lk
        print(f'x1 = {x1}')
        print(f'x2 = {x2}')
        fx1 = f(x1)
        fx2 = f(x2)
        print(f'f(x1) = {fx1}')
        print(f'f(x2) = {fx2}')
        if fx1 == fx2:
            a = x1
            b = x2
        elif fx1 > fx2:
            b = x2
        k+=1
        print('The values of a: ',a)
        print('The values of b: ',b)
    print('-----------Answer------------')
    print(f'The interval in which minimum lies [{a},{b}]')

main()

File: fibonacci.cpp
#include<iostream>
#include<math.h>
#include<iomanip>
#include<stdio.h>
#include<conio.h>
using namespace std;

int fib(int n)
{
    int a = 0, b = 1, c, i;
    if( n == 0)
        return a;
    for(i = 2; i <= n; i++)
    {
       c = a + b;
       a = b;
       b = c;
    }
    return b;
}

float f(double x)
{
    return pow(x,3)*(x-3)*pow((x-6),4);  //enter function here
    
}
//in this question (4,7) is the uncertainity region and n=3
int main()
{
    double a,b;
    int n;
    int k=2;
    int L;
    double x1,x2,Lk;
    cout<<"Enter the limits\n";
    cin>>a>>b;
    fflush(stdin);
    cout<<"Enter the Number of Function Evaluvations\n";
    cin>>n;
    cout<<endl;
    L = (b-a);
    cout<<"The value of L is :"<<L<<endl;
    while(k<=n)
    { 
        cout<<"\n \n";
        cout<<"The "<<(k-1)<<"th Iteration :"<<endl;
        int I_0 = (n-k+1);
        int I_1 = (n+1);
        int F_0 = fib(I_0+1);
        int F_1 = fib(I_1+1);
        Lk = ((double)F_0/(double)F_1)*L;
        cout<<" The Value of I_0 :"<<I_0<<endl;
        cout<<" The Value of I_1 :"<<I_1<<endl;
        cout<<" The Value of F_0 :"<<F_0<<endl;
        cout<<" The Value of F_1 :"<<F_1<<endl;
        cout<<" The Value of L_"<<k<<"  is :"<<Lk<<endl;
        x1 = a + Lk;
        x2 = b - Lk;
        cout<<"x1 :"<<x1<<endl;
        cout<<"x2 :"<<x2<<endl;
        cout<<" f(x1) : "<<f(x1)<<"\n f(x2) : "<<f(x2)<<endl;
        if(f(x1) == f(x2))
        {
            a = x1;
            b = x2;
        }
        else if(f(x1)>f(x2))
        {
            a = x1;
        }
        else
        {
            b = x2;
        }
        k++;
        cout<<"\n The Values of a : "<<a<<"\n The Values of b :"<<b<<endl;
    }

    cout<<"\n The Interval in which minimum lies ["<<a<<","<<b<<"] .";
}
